components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -112,31 +188,146 @@ function NavigationListItem(item: NavItem, props: DrawerNavGroupProps): ReactNod
         }
     };
 
+    const chevron = groupChevron === undefined ? itemChevron : groupChevron;
+
+    function getRightActionIcon(): JSX.Element {
+        if (item.rightComponent) {
+            return item.rightComponent;
```

Comment:
custom rightComponents still get transformed on click, even things like labels, etc.

Reply:
Let me hold on to these right component thing and talk to Kyle...